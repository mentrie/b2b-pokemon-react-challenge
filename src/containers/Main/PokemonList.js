import React from 'react';
import PropTypes from 'prop-types';
import PokemonItem from './PokemonItem';

const PokemonList = ({ pokemons, onSelect }) => {
  const renderPokemons = () => {
    return pokemons.map(pokemon => (
      <PokemonItem key={pokemon.name}
        pokemon={pokemon}
        onSelect={onSelect}
      />
    ));
  };

  return (
    <div className="main__pokemon__wrapper">
      {renderPokemons()}
    </div>
  );
};

PokemonList.propTypes = {
  pokemons: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    url: PropTypes.string.isRequired
  }))
};

export default PokemonList;
