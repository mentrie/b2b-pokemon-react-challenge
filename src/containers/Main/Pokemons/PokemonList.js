import React from 'react';
import PropTypes from 'prop-types';
import PokemonItem from './PokemonItem';
import Link from 'react-router-dom/Link';
import { getPokemonId } from '../../../helpers/utils';

const PokemonList = ({ pokemons, onSelect }) => {
  const renderPokemons = () => {
    return pokemons.map((pokemon, index) => (
      <Link
        to={`/${pokemon.name}/${getPokemonId(pokemon.url)}`}
        key={index}
        className="pokeman__list__item pokemon__no__link"
        onClick={onSelect}
      >
        <PokemonItem pokemon={pokemon} />
      </Link>
    ));
  };

  return (
    <div className="pokemon__list">
      {renderPokemons()}
    </div>
  );
};

PokemonList.propTypes = {
  pokemons: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    url: PropTypes.string.isRequired
  }))
};

export default PokemonList;
