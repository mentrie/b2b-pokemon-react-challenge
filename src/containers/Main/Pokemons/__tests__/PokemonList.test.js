import React from 'react';
import PokemonList from '../PokemonList';

test('render component', () => {
  const props = {
    pokemons: [{
      name: 'test',
      url: 'test/2'
    }]
  };

  const component = shallow(<PokemonList {...props} />);
  expect(component).toMatchSnapshot();
  expect(component.find('Link')).toBeDefined();
  expect(component.props().className).toEqual('pokemon__list');
});
