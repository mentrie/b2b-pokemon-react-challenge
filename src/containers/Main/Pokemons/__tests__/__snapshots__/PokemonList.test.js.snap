// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component 1`] = `
<div
  className="pokemon__list"
>
  <Link
    className="pokeman__list__item pokemon__no__link"
    key="0"
    replace={false}
    to="/test/NaN"
  >
    <PokemonItem
      pokemon={
        Object {
          "name": "test",
          "url": "test/2",
        }
      }
    />
  </Link>
</div>
`;
